engine/src/test/java/io/zeebe/engine/util/EngineRule.java

```diff
@@ -155,6 +159,13 @@ private void startProcessors() {
                           (key, partition) -> {},
                           jobsAvailableCallback)
                       .withListener(new ProcessingExporterTransistor()));
+
+          // sequenialize the commands to avoid concurrency
+          final var subscriptionHandlerExecutor = Executors.newSingleThreadExecutor();
```

Comment:
Here you create now an executor for each handler? What if we create one for all? So outside of the loop?

Reply:
It would be possible too. The handlers may block each other a bit. But I guess that it is not a big difference in your test setup.

Reply:
@Zelldon, I applied your suggestion and use only one thread pool for all subscription handler.